test/unit/BPool.t.sol

```diff
@@ -109,6 +113,28 @@ abstract contract BasePoolTest is Test, BConst, Utils, BMath {
     uint256 _foo = bpow(_y, _weightRatio);
     vm.assume(bsub(_foo, BONE) < type(uint256).max / _tokenInBalance);
   }
+
+  function _assumeCalcPoolOutGivenIn(
+    uint256 _tokenInDenorm,
+    uint256 _tokenInBalance,
+    uint256 _tokenAmountIn,
+    uint256 _swapFee,
+    uint256 _totalWeight,
+    uint256 _totalSupply
+  ) internal pure {
+    uint256 _normalizedWeight = bdiv(_tokenInDenorm, _totalWeight);
+    vm.assume(_normalizedWeight < BONE); // TODO: why this? if the weights are between allowed it should be fine
+
+    uint256 _zaz = bmul(bsub(BONE, _normalizedWeight), _swapFee);
+    uint256 _tokenAmountInAfterFee = bmul(_tokenAmountIn, bsub(BONE, _zaz));
+    uint256 _newTokenBalanceIn = badd(_tokenInBalance, _tokenAmountInAfterFee);
+    vm.assume(_newTokenBalanceIn < type(uint256).max / BONE);
+    vm.assume(_newTokenBalanceIn > _tokenInBalance);
```

Comment:
i sense that we could transform this on an internal as the other internal computing methods

Reply:
why is that?

Reply:
because `zaz` is not present at the pool, but on `calcPoolOutGivenSingleIn`, in similar methodology as in `_assumeCalcSpotPrice`, we could internalize this calcs so that is readable that we're expecting that not to fail